packages/vats/test/test-gov-starting.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+import test from 'ava';
+import { createRequire } from 'module';
+import fsp from 'fs/promises';
+
+import { permit } from '../src/core/gov-starting.js';
+
+const myRequire = createRequire(import.meta.url);
+
+/**
+ * If $SCRIPT is set, the test(s) below will write the rendered script to a file
+ * by that name. Likeiwse $PERMIT.
+ */
+test.before(async t => {
+  t.context = {
+    readFile: fsp.readFile,
+    env: process.env,
+    writeFile: fsp.writeFile,
+  };
+});
+
+test('check / save gimix permit', async t => {
+  t.notThrows(() => JSON.parse(permit));
+
+  // IDEA: check permit against a pattern
+
+  // @ts-expect-error TODO: types of t.context
+  const { env } = t.context;
+  if (env.PERMIT) {
+    // @ts-expect-error TODO: types of t.context
+    const { writeFile } = t.context;
+    const pretty = JSON.stringify(JSON.parse(permit), null, 2);
+    t.log('write permit to', env.PERMIT);
+    await writeFile(env.PERMIT, pretty);
+  }
+});
+
+const redactImportDecls = txt =>
+  txt.replace(/^\s*import\b\s*(.*)/gm, '// XMPORT: $1');
+const omitExportKewords = txt => txt.replace(/^\s*export\b\s*/gm, '');
+// cf. ses rejectImportExpressions
+// https://github.com/endojs/endo/blob/ebc8f66e9498f13085a8e64e17fc2f5f7b528faa/packages/ses/src/transforms.js#L143
+const hideImportExpr = txt => txt.replace(/\bimport\b/g, 'XMPORT');
+
+test('module to script: redact imports; omit export keywords', t => {
+  const modText = `
+import { E, Far } from '@endo/far';
+
+/** @param {import('wonderland').Carol} carol */
+export contst alice = (carol) => {
+    E(bob).greet(carol);
+};
+      `;
+
+  const expected = `
+// XMPORT: { E, Far } from '@endo/far';
+
+/** @param {XMPORT('wonderland').Carol} carol */
+contst alice = (carol) => {
+    E(bob).greet(carol);
+};
+      `;
+
+  const script = hideImportExpr(redactImportDecls(omitExportKewords(modText)));
+  t.is(script.trim(), expected.trim());
+});
+
+test('check / render gov-starting proposal', async t => {
+  // @ts-expect-error TODO: types of t.context
+  const { env, readFile } = t.context;
+
+  const modText = await readFile(
+    myRequire.resolve('../src/core/gov-starting.js'),
+    'utf-8',
+  );
+
+  let script = hideImportExpr(redactImportDecls(omitExportKewords(modText)));
```

Comment:
please factor this into a single function, with the explanation. consider giving that function a separate module and test file